src/compiler/checker.ts

```diff
@@ -2724,15 +2724,13 @@ namespace ts {
                     const anyImportSyntax = getAnyImportSyntax(declaration);
                     if (anyImportSyntax &&
                         !hasModifier(anyImportSyntax, ModifierFlags.Export) && // import clause without export
-                        isDeclarationVisible(<Declaration>anyImportSyntax.parent)) {
-                        // In function "buildTypeDisplay" where we decide whether to write type-alias or serialize types,
-                        // we want to just check if type- alias is accessible or not but we don't care about emitting those alias at that time
-                        // since we will do the emitting later in trackSymbol.
-                        if (shouldComputeAliasToMakeVisible) {
-                            getNodeLinks(declaration).isVisible = true;
-                            aliasesToMakeVisible = appendIfUnique(aliasesToMakeVisible, anyImportSyntax);
-                        }
-                        return true;
+                        isDeclarationVisible(anyImportSyntax.parent)) {
+                        return addVisibleAlias(declaration, anyImportSyntax);
+                    }
+                    else if (isVariableDeclaration(declaration) && isVariableStatement(declaration.parent.parent) &&

```

Comment:
should not we do this for all declarations and not jsut  variable declarations? e.g. https://github.com/Microsoft/TypeScript/issues/23127

Reply:
So allow private interfaces, types, anything - to be marked and emitted?

Reply:
why are variables any special? they are all declarations..

Reply:
Mostly just because they're only used to name unique symbols. I can update to handle the other kinds of declaration statements, but it'll require more work to light up each private declaration kind in the declaration emitter.

Reply:
Currently with your change, making variable statement visible will make all declarations in it visible.. Thats not what you are intending i suppose.

Reply:
Also did @mhegazy's ask included making it visible all the time.. I thought it implied only in ```export =``` case i think, because other wise we would be making lot of stuff visible which might not be intended?

Reply:
he added code later on to make it emit an extra `export {};` at the bottom, this should change the module scoping.

Reply:
can you add a test wherein only one variable declaration becomes visible. Thanks.

Reply:
@sheetalkamat done